tidy3d/plugins/smatrix/smatrix.py

```diff
@@ -92,32 +90,141 @@ class ComponentModeler(Tidy3dBaseModel):
         title="Folder Name",
         description="Name of the folder for the tasks on web.",
     )
-
-    batch: Tidy3dBaseModel = pd.Field(
+    element_mappings: Dict[Element, Dict[Element, Callable[[complex], complex]]] = pd.Field(
+        {},
+        title="Element Mappings",
+        description="Mapping between elements of the scattering matrix, as specified by"
+        "specified by pairs of ``(port name, mode index)`` matrix indices. "
+        "``element_mappings[element1][element2]`` returns a function ``f`` that sets the matrix "
+        "element2 as ``f(element1)``. "
+        "Each ``element`` is defined by a pair of indices ``((str, int), (str, int))`` relating "
+        "the source port name and mode index to the monitor port name and mode index.",
+    )
+    run_only: Optional[Tuple[MatrixIndex, ...]] = pd.Field(
+        None,
+        title="Run Only",
+        description="If specified, a tuple of matrix indices, specified by (:class:`.Port`, `int`),"
```

Comment:
``int``?

I guess we need to try to keep track of this / check in PRs: in docstrings, RST. In error messages - simple ' '

Reply:
yea. I'll fix any more if I see them.